Finjector.Core/Services/AggieEnterpriseService.cs

```diff
@@ -0,0 +1,273 @@
+﻿using Finjector.Core.Models;
+using Microsoft.Extensions.Options;
+using AggieEnterpriseApi;
+using AggieEnterpriseApi.Extensions;
+using AggieEnterpriseApi.Validation;
+
+
+namespace Finjector.Core.Services
+{
+    public interface IAggieEnterpriseService
+    {
+        //Returns Chart type (includes invalid value)
+        FinancialChartStringType GetChartType(string segmentString);
+
+        Task<AeDetails> GetAeDetailsAsync(string segmentString);
+    }
+
+    public class AggieEnterpriseService : IAggieEnterpriseService
+    {
+        private readonly FinancialOptions _financialOptions;
+        private IAggieEnterpriseClient _apiClient;
+
+        public AggieEnterpriseService(IOptions<FinancialOptions> options)
+        {
+            _financialOptions = options.Value;
+            _apiClient = GraphQlClient.Get(_financialOptions.ApiUrl!, _financialOptions.TokenEndpoint!, _financialOptions.ConsumerKey!, _financialOptions.ConsumerSecret!, $"{_financialOptions.ScopeApp}-{_financialOptions.ScopeEnv}");
+        }
+
+        public async Task<AeDetails> GetAeDetailsAsync(string segmentString)
+        {
+            AeDetails aeDetails = new AeDetails();
+            aeDetails.ChartType = string.IsNullOrEmpty(segmentString) ? FinancialChartStringType.Invalid.ToString() : GetChartType(segmentString).ToString().ToUpper();
+            if(aeDetails.ChartType == FinancialChartStringType.Invalid.ToString())
+            {
+                aeDetails.Errors.Add("Invalid Chart Type");
+                aeDetails.IsValid = false;
+                return aeDetails;
+            }
+
+            if(aeDetails.ChartType == FinancialChartStringType.Gl.ToString().ToUpper())
+            {
+                var glSegments = FinancialChartValidation.GetGlSegments(segmentString);
+                var result = await _apiClient.DisplayDetailsGl.ExecuteAsync(
+                    segmentString: segmentString, validateCVRs: true, 
+                    project: glSegments.Project, 
+                    entity: glSegments.Entity, 
+                    fund: glSegments.Fund, 
+                    dept: glSegments.Department, 
+                    account: glSegments.Account, 
+                    purpose: glSegments.Purpose, 
+                    program: glSegments.Program, 
+                    activity: glSegments.Activity);
+
+                var data = result.ReadData();
+
+                aeDetails.IsValid = data.GlValidateChartstring.ValidationResponse.Valid;

```

Comment:
sometimes you use `data` and sometimes `data?.` -- seems like nulls and not nulls are being mixed up.  Perhaps just throw in a `if (data == null) {... }` and then you'd have a lot less to worry about and things would be a bit easier to read